R/plotting.R

```diff
@@ -0,0 +1,282 @@
+
+#' Crunch ggplot theme
+#' 
+#' Style ggplots according to Crunch style.
+#'
+#' @param base_size Base text size
+#' @param base_family Base text family
+#'
+#' @export
+#' 
+#' @importFrom ggplot2 element_line element_blank element_text element_rect 
+#' theme_minimal theme  rel unit 
+theme_crunch <- function(base_size = 12, base_family = "sans") {
+    subtitle <- element_text(
+                    hjust = 0, 
+                    face = "bold",
+                    size = rel(1), 
+                    color = card_colors[1])
+    (theme_minimal() +
+            theme(
+                line = element_line(colour = "black"),
+                axis.title = element_blank(),
+                axis.ticks = element_blank(),
+                axis.line = element_blank(),
+                panel.grid.minor = element_blank(),
+                plot.title = element_text(
+                    hjust = 0, 
+                    size = rel(1.5),
+                    face = "bold", 
+                    color = card_colors[2]),
+                plot.subtitle = subtitle,
+                plot.margin = unit(c(1, 1, 1, 1), "lines"),
+                legend.title = subtitle,
+                strip.background = element_rect(fill = "light gray", linetype = 0))
+    )
+}
+
+crunch_colors <- c(
+    "#316395", "#cf3e3e", "#fcb73e", "#37ad6c", "#9537b5", "#17becf", 
+    "#e377c2", "#fdae6b", "#0099c6", "#ed5487", "#3366cc"
+    )
+
+card_colors <- c("#56A08E","#007F65")
+
+#' @importFrom viridisLite viridis
+#' @importFrom dplyr summarize pull
+generate_colors <- function(var) {
+    n_cols <- length(unique(var))
+    if (n_cols > length(crunch_colors)) {
+        return(c(crunch_colors, viridis(n_cols - length(crunch_colors))))
+    } else {
+        return(crunch_colors)
+    }
+}
+
+
+#' @rdname autoplot
+#' @importFrom ggplot2 aes autoplot geom_histogram ggplot labs
+#' @importFrom crunch description name
+#' @export
+autoplot.DatetimeVariable <- function(x, ...) {
+    v <- as.vector(x)
+    plot_df <- data_frame(!!sym(name(x)) := as.Date(as.vector(x)))
+
+    ggplot(plot_df, aes(x = !!sym(name(x)))) + 
+        geom_histogram(fill = card_colors[2]) + 
+        theme_crunch() +
+        labs(title = name(x),
+            subtitle = description(x))
+}
+
+#' @rdname autoplot
+#' @importFrom ggplot2 aes autoplot geom_histogram ggplot labs
+#' @importFrom crunch description name
+#' @importFrom rlang !! sym :=
+#' @export
+autoplot.NumericVariable <- function(x, ...) {
+    v <- as.vector(x)
```

Comment:
The logic for including the full vector so that they can change the binwidth after the fact. So if they wanted to do: 
```
autoplot(ds$var) +
    geom_hisogram(binwidth = 4)
```
Then the numeric vector needs to be in the ggplot object.